src/v/serde/serde_is_enum.h

```diff
@@ -17,7 +17,7 @@ using serde_enum_serialized_t = int32_t;
 
 template<typename T>
 inline constexpr bool serde_is_enum_v =
-#if __has_cpp_attribute(__cpp_lib_is_scoped_enum)
+#if defined(__cpp_lib_is_scoped_enum) && __cpp_lib_is_scoped_enum >= 202011L
   std::is_scoped_enum_v<T>
   && sizeof(std::decay_t<T>) <= sizeof(serde_enum_serialized_t);
 #else
```

Comment:
why do we need the feature check if this is available in c++23? is it the case that it isn't in libc++?

Reply:
Because vtools is still c++20

Reply:
https://github.com/redpanda-data/vtools/pull/3362

Reply:
makes sense